nfs/src/rpc_task.cpp

```diff
@@ -1313,30 +1273,49 @@ void rpc_task::run_write()
      * membufs. We don't wait for the writes to actually finish, which means
      * we support buffered writes.
      */
-    std::vector<bytes_chunk> bc_vec =
-            copy_to_cache(this, ino, bufv, offset, length, error_code);
+    uint64_t periodic_bytes = 0;
+
+    copy_to_cache(this, ino, bufv, offset, length, error_code, &extent_left, &extent_right);
     if (error_code != 0) {
         AZLogWarn("[{}] copy_to_cache failed with error={}, "
                   "failing write!", ino, error_code);
         reply_error(error_code);
         return;
     }
 
-    assert(bc_vec.size() > 0);
+    assert(extent_right >= extent_left);
 
-    for (bytes_chunk& bc : bc_vec) {
-        /*
-         * Create the flush task to carry out the write.
-         */
-        struct rpc_task *flush_task =
-            get_client()->get_rpc_task_helper()->alloc_rpc_task(FUSE_FLUSH);
-        flush_task->init_flush(nullptr /* fuse_req */, ino);
+    /*
+     * Get the prune goals, if it's more than data already in-flight
+     * data to BLOB, then issue fresh flush.
+     */
+    inode->filecache_handle->get_prune_goals(nullptr, &periodic_bytes);
+    AZLogDebug("periodic_bytes: {}, extent_left{}, extent_right: {}",periodic_bytes, extent_left, extent_right);
+
+    if (inode->inflight_bytes >= periodic_bytes) {
```

Comment:
inflight_bytes (as the name suggests and also as commented above) is the write bytes we have dispatched and awaiting response. It doesn't look right to compare that with periodic_bytes, as periodic_bytes is the amount of dirty data that we can hold in cache before we have to flush (periodically). if something is already dispatched, then why do we compare that with dirty bytes.

Reply:
Actually inflight_bytes is neither exactly inflight nor exactly dirty. You increment it when you add the iovector to flush but decrement when you get a write response.